curriculum/challenges/english/25-front-end-development/lecture-react-strategies-and-debugging/67d2f5dacd5e0c749e5d534c.md

```diff
@@ -2,13 +2,113 @@
+
+- View the app's component hierarchy.
+- Check and modify props, states, and context values in real time.
+- Check the source code for each selected component.
+- Log the component data to the console.
+- Inspect the DOM elements for the component.
+
+On the other hand, the Profiler tab helps you record and analyze component performance so you can identify unnecessary re-renders, view commit durations, and things you can optimize.
+
+Here's a simple app to show you how you can inspect components and any props and state they have. This is similar to the code we used in a previous lecture on prop drilling:
+
+```jsx
+import { useState } from "react";
+
+export default function App() {
+  const greeting = "Hello, Prop Drilling!";
+  const response = "I'm not here to play!";
+
+  return <Parent greeting={greeting} response={response} />;
+}
+
+const Parent = ({ greeting, response }) => {
+  return <Child greeting={greeting} response={response} />;
+};
+
+const Child = ({ greeting, response }) => {
+  return <Grandchild greeting={greeting} response={response} />;
+};
+
+const Grandchild = ({ greeting, response }) => {
+  const [count, setCount] = useState(0);
+
+  return (
+    <>
+      <h1>{greeting}</h1>
+      <h2>{response}</h2>
+
+      <button onClick={() => setCount(count + 1)}>Increase Count</button>
+      <h2>Count: {count}</h2>
+      <button onClick={() => setCount(count - 1)}>Decrease Count</button>
+    </>
+  );
+};
+```
+
+Here's the tree view of the components:
+
+[Image showing the tree view of the components in React DevTools, with the `App` component at the top, followed by `Parent`, `Child`, and `Grandchild` components]
+
+If you select any of these components, you can see the props and state in them:
+
+[Video clip showing the current props and state of the `Parent` component after it is clicked on]
+
+You can see the props and state update in real-time, and change them if necessary:
+
+[Video clip of selecting the `Child` component and updating the `greeting` prop from `Hello, Prop Drilling!` to `Hello, Welcome to Prop Drilling!`]
+
+To log data in a component to the console, inspect the matching DOM elements and view the source code of the component. The icons in the top right corner let you do that:
+
+[Video clip showing the `Log the component data to the console` button being clicked and logging the data in the `Grandchild` component to the console]
+
+A common bug you might encounter in React is called props mismatch.
+
+For example, say that for the `Child` component, you mistakenly pass in `reply` as the prop instead of `response`:
+
+```jsx
+const Child = ({ greeting, response }) => {
+  return <Grandchild greeting={greeting} reply={response} />;
+};
+```
+
+Remember that `Grandchild` expects a response prop. Because the component receives a different prop, it can't display that text on the page, and just adds and empty `h2` to the DOM:
```

Comment:
```suggestion
Remember that `Grandchild` expects a `response` prop. Because the component receives a different prop, it can't display that text on the page, and just adds and empty `h2` to the DOM:
```